client/main/vareditor/datavarwidget.js

```diff
@@ -128,19 +132,107 @@ const DataVarWidget = Backbone.View.extend({
             this._enableDisableMoveButtons();
         }
     },
-    _setLevels(levels) {
+    _setLevels(measureType, levels) {
         if ( ! this.attached)
             return;
-        this.$levelItems.off('click');
-        this.$levels.empty();
+
+        if (levels.length === 0)
+            this.$levels.empty();
+        else
+            this.$levelItems.remove(":gt(" + (levels.length - 1) + ")");
 
         this.$moveUp.addClass('disabled');
         this.$moveDown.addClass('disabled');
 
         if (levels) {
+
+            let keydownFunction = ((event) => {
+                let $input = $(event.delegateTarget);
+                let keypressed = event.keyCode || event.which;
+                if (keypressed === 13) { // enter key
+                    $input.blur();
+                    if (this.model.get('changes'))
+                        this.model.apply();
+                    event.preventDefault();
+                    event.stopPropagation();
+                }
+                else if (keypressed === 27) { // escape key
+                    $input.blur();
+                    if (this.model.get('changes'))
+                        this.model.revert();
+                    event.preventDefault();
+                    event.stopPropagation();
+                }
+            });
+
+            let changeFunction = (event) => {
+                let $input = $(event.delegateTarget);
+                let index = $input.data('index');
+                this.model.editLevelLabel(index, $input.val());
+                let level = this.model.get('levels')[index];
+                let diff = level.importValue !== level.label;
+                let $ivLabel = this.$levels.find(".silky-variable-editor-level[data-index = " + index + "]");
+                $ivLabel.attr('data-changed', diff);
+            };
+
+            let focusFunction = (event) => {
+                this._currentKeyboardContext = keyboardJS.getContext();
+                keyboardJS.setContext('');
```

Comment:
`keyboardjs.pause()` wouldn't work here? would save you from having to store and restore the old context.